compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -1452,7 +1453,7 @@ public void visit(BLangInvocation invocationExpr) {
         BSymbol symbol = invocationExpr.symbol;
         this.unusedLocalVariables.remove(symbol);
 
-        if (!isGlobalVarsInitialized(invocationExpr.pos)) {
+        if (symbol.owner == getEnclPkgSymbol(env) && !isGlobalVarsInitialized(invocationExpr.pos)) {
```

Comment:
Consider the following scenario
```ballerina
function init() {
    Foo foo;
    foo.value();
}

public function main() {
}

class Foo {
    function value() {
    }
}
```
Without this change, this gives an error as
```
ERROR [test3.bal:(6:5,6:8)] variable 'foo' is not initialized
ERROR [test3.bal:(6:5,6:16)] variable(s) 'foo' not initialized
```
With this change, this gives an error as
```
ERROR [test3.bal:(6:5,6:8)] variable 'foo' is not initialized
```
The reason for that is the `symbol.owner` of `foo.value()` is `Foo` and  `getEnclPkgSymbol` gives symbol of package.
IMO for this kind of scenario, this check cannot be false. because `value()` and `foo` both in the same package.
WDYT?

Reply:
For the first case, `foo` is a local variable. So the only error there is that `foo` is not initialized before calling `foo.value()`.

I think part of the issue is that the error wasn't clear. I've changed the error to better reflect the reason for the error. Please check.